app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/UpscanService.scala

```diff
@@ -62,81 +69,41 @@ class UpscanServiceImpl @Inject() (
 
   private val maxUploads: Int = getUpscanInitiateConfig[Int]("max-uploads")
 
-  override def storeNotifyEvent(cgtReference: CgtReference, upscanNotifyResponse: UpscanNotifyResponse)(
-    implicit hc: HeaderCarrier
-  ): EitherT[Future, Error, Unit] =
-    upscanStore
-      .insert(UpscanNotifyEvent(cgtReference.value, upscanNotifyResponse))
-      .leftMap { error =>
-        logger.warn(s"Could not store upscan notify event due to $error")
-        error
-      }
-
   override def initiate(
     cgtReference: CgtReference
-  )(implicit hc: HeaderCarrier): EitherT[Future, Error, UpscanServiceResponse] =
-    upscanStore.fileUploadCount(cgtReference).flatMap { uploadCount =>
-      if (uploadCount > maxUploads) {
-        EitherT.rightT(UpscanServiceResponse.MaximumFileUploadReached)
-      } else {
-        for {
-          response <- upscanConnector.initiate(cgtReference)
-          descriptor <- EitherT.fromOption(
-                         response.json.validate[UpscanDescriptor].asOpt,
-                         Error("S3 Json payload structure is not as expected")
-                       )
-        } yield UpscanResponse(cgtReference.value, descriptor)
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, UpscanInitiateResponse] =
+    upscanConnector.getUpscanSnapshot(cgtReference).flatMap {
+      case Some(upscanSnapshot) => {
+        if (upscanSnapshot.fileUploadCount > maxUploads) {
+          EitherT.rightT(MaximumFileUploadReached)
+        } else {
+          for {
+            response <- upscanConnector.initiate(cgtReference)
```

Comment:
can any of this be put into the BE?

Reply:
No, not really. This is the bit when the FE makes a call to the `upscan-initiate` service. The file upload form has not been rendered to the user yet. Once a successful response has been received by `upscan-initiate` we *have* to successfully store the file descriptors *before* we can render the form to the user. If we don't make this constraint then potentially we could show a file upload form but there are no file descriptors in the database when it comes to uploading the form to S3.